src/components/app/ProfileSharing.js

```diff
@@ -62,6 +64,16 @@ UploadingStatus.propTypes = {
 };
 
 class ProfileSharingCompositeButton extends PureComponent {
+  _attemptToShare: Function;
+  _onPermalinkPanelOpen: Function;
+  _onPermalinkPanelClose: Function;
+  _permalinkButtonCreated: Function;
+  _permalinkButton: Object;
+  _uploadErrorButtonCreated: Function;
+  _uploadErrorButton: Object;
+  _permalinkTextFieldCreated: Function;
+  _permalinkTextField: Object;
```

Comment:
So, we can do better here :)

1. For the functions we bind, we use code like `(this: any)._attemptToShare = this._attemptToShare.bind(this);` so that we don't have to manually add properties that are mostly useless.

2. For the other functions, we try to bind them better. For example `_permalinkButtonCreated` is `ButtonWithPanel => void`.

3.  The properties like `_permalinkButton` needs to be better typed too. `_permalinkButton` and `_uploadErrorButton` are `?ButtonPanel` and `_permalinkTextField` is a `?HTMLInputElement`.

4. and as said above, please type the props, the state, and the methods :)